src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -492,12 +492,19 @@ public function compileSpatialIndex(Blueprint $blueprint, Fluent $command)
      */
     protected function compileKey(Blueprint $blueprint, Fluent $command, $type)
     {
-        return sprintf('alter table %s add %s %s%s(%s)',
+        $columns = collect($command->columns)->map(function ($column) {
```

Comment:
Would be nice to type-hint `$column`:
```suggestion
        $columns = collect($command->columns)->map(function (string $column) {
```
you might want to add `Stringable` as well, but I'm not sure if that is needed